core/src/dird/scheduler_private.cc

```diff
@@ -127,7 +127,6 @@ static void SetJcrFromRunResource(JobControlRecord* jcr, RunResource* run)
 JobControlRecord* SchedulerPrivate::TryCreateJobControlRecord(
     const SchedulerJobItem& next_job)
 {
-  if (next_job.job->client == nullptr) { return nullptr; }
   JobControlRecord* jcr = NewDirectorJcr();
   SetJcrDefaults(jcr, next_job.job);
```

Comment:
`SetJcrDefaults()` already has checks for `job->client` in place, so it should be safe to assume it may be `nullptr`.

Reply:
`SetJcrFromRunResource()` does not touch `job->client` at all

Reply:
The only additional requirement seems to be that `ExecuteJobCallback_()` needs to be able to handle a jcr with no client set, which is an assumption we can probably make.